contracts/Splitter.sol

```diff
@@ -44,22 +48,24 @@ contract Splitter is Pausable {
     }
 
     function withdraw(uint amount) public whenNotPaused returns(bool) {
-        uint256 msgSender = balances[msg.sender];
+        uint256 senderBalance = balances[msg.sender];
         require(amount > 0, "The value must be greater than 0");
-        require(msgSender >= amount, "There are insufficient funds");
+        require(senderBalance >= amount, "There are insufficient funds");
 
-        balances[msg.sender]  = SafeMath.sub(msgSender, amount);
+        balances[msg.sender]  = SafeMath.sub(senderBalance, amount);
         emit WithDraw(msg.sender, amount);
         (bool success, ) = msg.sender.call{value: amount}("");
         require(success, "Transfer failed");
         return true;
     }
 
     function pause() public {
+        require(msg.sender == owner, "The contract can only be paused by the owner");
         super._pause();
     }
 
     function unpause() public {
+        require(msg.sender == owner, "The contract can only be unpaused by the owner");
```

Comment:
You can take out this small require into its own modifier. It is a classic pattern for owner in Solidity.

Reply:
I have extended from Ownable.sol